### DISCUSSION TITLE
Equivalent of screen -R ?

### DISCUSSION THREAD

**Discussion by osennte** [tmux/tmux#3920] [General]
Hello, with the old screen program I frequently used the -R option: With it I did not need to check if a screen session with the given name exists, because the program connected to it if present, or created it and connected if not. I failed to find a simple command line with tmux that behaves this way "idempotently" (with simple I mean without: if tmux ls ...; then attach...; else new ...; fi). Please help!

**Comment by aless3** [comment]
what about
```tmux new-session -A -s NAME```?
NAME is the name of the session, by default 0.
this can be shortened to ```tmux new -As0``` if using the default session name, or changing the 0 the connect a specific tmux session (creating it if not already existing)
in a small test I just did it also seems that ```tmux new -A``` works if using only one session (haven't tested with more sessions)
I never used screen, but it seems to me that "-As" should be the equivalent of screen -R

A.

**Reply by osennte** [reply]
Yes that does the trick, thanks!

**Reply by anarcat** [reply]
is there a way to provide a short-hand alias for this like `tmux A`? i know how this sounds, but `tmux new -A` is quite long compared to `screen -RD` which is pretty tightly hardwired in my fingers.

i've tried the mysterious `command-alias` but this doesn't seem to do anything:

```
set command-alias[0] A="new-session -A"
```

it fails with `no server running on /tmp/tmux-1000/default`.

as for this:

>  I never used screen, but it seems to me that "-As" should be the equivalent of screen -R

Yes, `tmux new-session -A` is pretty similar to `screen -R`. Here is what my hardwired `-RD` does:

```
       -D -R   Attach  here  and now. In detail this means: If a session is run‐
               ning, then reattach. If  necessary  detach  and  logout  remotely
               first.  If it was not running create it and notify the user. This
               is the author's favorite.
```

`-D` is mostly irrelevant for tmux, as tmux happily reattaches to an existing session even if it's already attached elsewhere. In screen, you need to first detach the other with `-D` or use `-x` to have two clients attached to the same session.

`-R`, essentially, means "attach or create".

**Reply by aless3** [reply]
I'm not sure if `-A` has everything or not (I'm not a native speaker).

In any case I would suggest a simple alias: `alias ta="tmux new -A"`, in your bashrc/equivalent (change the command if you need something different) 
no messing with subcommands and the like. 
easier, shorter, probably not as clean, but I find myself happy with this kind of aliases (like tk instead of tmux kill-server, g instead of git etc)
just keep in mind to not override built in commands or other functions/packages, this would lead to headaches down the line

**Reply by anarcat** [reply]
yeah, everyone essentially says "make a bash alias or function", i think that's sub-par. i'd much rather see a function for this inside tmux, and upstream, personally, as i manage a lot of servers and propagating such policy changes everywhere is a pain in the back.

**Reply by nicm** [reply]
We're not going to add something shorter than "tmux new -A" because you can't get used to typing 11 characters instead of 10, be serious.

**Comment by anarcat** [comment]
> We're not going to add something shorter than "tmux new -A" because you can't get used to typing 11 characters instead of 10, be serious.

I was thinking more something likee `tmux A`, 6 characters, and I'm
quite serious, but I understand where you're coming from. :)
